dspace/cli/senior_thesis_community.rb

```diff
@@ -0,0 +1,354 @@
+module DSpace
+  module CLI
+    class SeniorThesisCollection
+      attr_reader :obj
+
+      def self.kernel
+        ::DSpace
+      end
+
+      def initialize(obj)
+        @obj = obj
+      end
+
+      def removeItem(item)
+        @obj.removeItem(item.obj)
+      end
+
+      def update
+        @obj.update
+        self.class.kernel.commit
+      end
+    end
+
+    class SeniorThesisWorkflowItem
+      attr_reader :obj
+
+      def self.kernel
+        ::DSpace
+      end
+
+      # This is some internal bug; I am not certain why I cannot use this from DItem
+      def initialize(obj)
+        @obj = obj
+      end
+
+      def delete
+        @obj.deleteWrapper
+        self.class.kernel.commit
+        @obj = nil
+      end
+    end
+
+    class SeniorThesisItem < ::DItem
+      java_import org.dspace.workflow.WorkflowItem
+      java_import org.dspace.content.Metadatum
+
+      attr_reader :obj
+
+      def self.kernel
+        ::DSpace
+      end
+
+      # This is some internal bug; I am not certain why I cannot use this from DItem
+      def initialize(obj)
+        @obj = obj
+      end
+
+      # This is some internal bug; I am not certain why I cannot use this from DItem
+      def getMetadataByMetadataString(metadata_field)
+        @obj.getMetadataByMetadataString(metadata_field)
+      end
+
+      def self.build_metadata(schema, element, qualifier = nil, language = nil)
+        metadata = Java::OrgDspaceContent::Metadatum.new
+        metadata.schema = schema
+        metadata.element = element
+        metadata.qualifier = qualifier
+        metadata.language = language
+        metadata
+      end
+
+      def get_metadata
+        @obj.getMetadata
+      end
+
+      def metadata
+        list = Utils::ArrayList.new(get_metadata)
+        list.to_a
+      end
+
+      def save
+        @obj.save
+        self.class.kernel.commit
+        self
+      end
+
+      def add_metadata(schema, element, value, qualifier = nil, language = nil)
+        metadata = self.class.build_metadata(schema, element, qualifier, language)
+        metadata.value = value
+        updated = get_metadata
+        get_metadata.add(metadata)
+        set_metadata(updated)
+        metadata
+      end
+
+      class Java::OrgDspaceContent::DSpaceObject::MetadataCache
```

Comment:
This is extremely problematic, as I cannot readily access https://github.com/DSpace/DSpace/blob/dspace-5.6/dspace-api/src/main/java/org/dspace/content/DSpaceObject.java#L1254 and manipulate the underlying metadata data entries. One should be able to use `field_accessor` (as outlined in https://github.com/jruby/jruby/wiki/CallingJavaFromJRuby), but this raises an error. As a consequence, I cannot, using JRuby update metadata for Items.